src/synapse/api.py

```diff
@@ -149,3 +151,35 @@ def _get_nonce() -> str:
     ) as exc:
         logger.error("Failed to request %s : %s", REGISTER_URL, exc)
         raise NetworkError(f"Failed to request {REGISTER_URL}.") from exc
+
+
+def get_version() -> str:
+    """Get version.
+
+    Expected API output:
+    {
+        "server_version": "0.99.2rc1 (b=develop, abcdef123)",
+        "python_version": "3.7.8"
+    }
+
+    Returns:
+        The version returned by Synapse API.
+
+    Raises:
+        NetworkError: if there was an error fetching the version.
+    """
+    try:
+        res = requests.get(VERSION_URL, timeout=5)
+        res.raise_for_status()
+        server_version = res.json()["server_version"]
+        version_match = re.search(r"^([^\s(]+)", server_version)
+        if version_match:
+            return version_match.group(1)
+        return server_version
+    except (
+        requests.exceptions.ConnectionError,
+        requests.exceptions.Timeout,
+        requests.exceptions.HTTPError,
+    ) as exc:
+        logger.error("Failed to request %s : %s", VERSION_URL, exc)
```

Comment:
When logging exceptions, using `%s` only print the message part for most of exception, without the exception type.

```
print("%s" % RuntimeError("some error")) # print some error
print("%r" % RuntimeError("some error")) # print RuntimeError('some error')
```

But it will be even better if we use `logger.exception` which will print the traceback of the exception as well.

Reply:
Great suggestion! Changed, thanks